src/symbolic/symbolic_choice.ml

```diff
@@ -71,31 +71,32 @@ module CoreImpl = struct
     let worker_local : ('a, 'a) t = Sched (fun wls -> Now wls)
   end
 
-  module Scheduler = struct
+  module Scheduler (Work_datastructure : Work_ds_intf.S) = struct
     (*
         A scheduler for Schedulable values.
       *)
-    type ('a, 'wls) work_queue = ('a, 'wls) Schedulable.t Wq.t
+    type ('a, 'wls) work_queue = ('a, 'wls) Schedulable.t Work_datastructure.t
 
     type ('a, 'wls) t = { work_queue : ('a, 'wls) work_queue } [@@unboxed]
 
     let init_scheduler () =
-      let work_queue = Wq.make () in
+      let work_queue = Work_datastructure.make () in
       { work_queue }
 
-    let add_init_task sched task = Wq.push task sched.work_queue
+    let add_init_task sched task =
+      Work_datastructure.push task 0 sched.work_queue
 
     let work wls sched callback =
       let rec handle_status (t : _ Schedulable.status) write_back =
         match t with
         | Stop -> ()
         | Now x -> callback x
-        | Yield (_prio, f) -> write_back f
+        | Yield (prio, f) -> write_back (Prio.to_int prio, f)
```

Comment:
I think it would be better if this `Prio.to_int` was made inside `Wq,Wpq,Ws` instead.